### ISSUE TITLE
InitializeOnStartup fails silently if Authorize attribute with ApplyPolicy.Validation is used

### ISSUE THREAD

**Issue by samisq** [ChilliCream/graphql-platform#5765]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Product

Hot Chocolate

### Describe the bug

InitializeOnStartup fails silently if `[Authorize]` attribute with `ApplyPolicy.Validation` is applied on an entity. The error is: `The feature Authorization is missing the state with the key HottChocolate.Authorization.UserState`.  The exception doesn't prevent startup, but it may cause HotChocolate to not be fully "warmed up".

### Steps to reproduce

1. Using OOB HotChocolate template, annonate `Book` entity with `Authorize` attribute like:
```
[HotChocolate.Authorization.Authorize(Apply = ApplyPolicy.Validation)]
public record Book(string Title, Author Author);
```

2. Add the following diagnostic listener so that we can see the exception (otherwise, it's swallowed)
```
public class LoggingExecutionEventListener : ExecutionDiagnosticEventListener
{
    private readonly ILogger<LoggingExecutionEventListener> _logger;
    public LoggingExecutionEventListener(ILogger<LoggingExecutionEventListener> logger) => _logger = logger;
    public override void RequestError(IRequestContext context, Exception exception)
        => _logger.LogError(exception, "An error occurred during GraphQL request execution");
}
```
3. Update `Program.cs` with the following code:
```
var builder = WebApplication.CreateBuilder(args);
builder.Services.AddAuthorization();
builder.Services
    .AddGraphQLServer()
    .AddAuthorization()
    .AddDiagnosticEventListener<LoggingExecutionEventListener>()
    .AddTypes()
    .InitializeOnStartup();

var app = builder.Build();
app.MapGraphQL();
app.Run();
```
 4. Run the app

### Relevant log output

```shell
fail: example1.Types.LoggingExecutionEventListener[0]
      An error occurred during GraphQL request execution
      HotChocolate.MissingStateException: The feature `Authorization` is missing the state with the key `HottChocolate.Authorization.UserState`.
         at HotChocolate.AspNetCore.Authorization.DefaultAuthorizationHandler.GetUserState(IDictionary`2 contextData)
         at HotChocolate.AspNetCore.Authorization.DefaultAuthorizationHandler.AuthorizeAsync(AuthorizationContext context, IReadOnlyList`1 directives, CancellationToken ct)
         at HotChocolate.Authorization.AuthorizeValidationResultAggregator.AggregateAsync(IDocumentValidatorContext context, DocumentNode document, CancellationToken ct)
         at HotChocolate.Validation.DocumentValidator.RunResultAggregators(DocumentValidatorContext context, DocumentNode document, CancellationToken ct)
         at HotChocolate.Execution.Pipeline.DocumentValidationMiddleware.InvokeAsync(IRequestContext context)
         at HotChocolate.Execution.Pipeline.DocumentParserMiddleware.InvokeAsync(IRequestContext context)
         at HotChocolate.Execution.Pipeline.DocumentCacheMiddleware.InvokeAsync(IRequestContext context)
         at HotChocolate.Execution.Pipeline.TimeoutMiddleware.InvokeAsync(IRequestContext context)
         at HotChocolate.Execution.Pipeline.ExceptionMiddleware.InvokeAsync(IRequestContext context)
```

### Additional Context?

One potential fix, update `ExecutorWarmupService` to include a stub for `UserState` in the query request, like:

```
// initialize pipeline with warmup request
            IQueryRequest warmupRequest = QueryRequestBuilder.New()
                .SetQuery("{ __typename }")
                .SetGlobalState(HotChocolate.WellKnownContextData.UserState, new UserState(new ClaimsPrincipal())); // <-- Add this
                .AllowIntrospection()
                .Create();
```

### Version

13.0.0-rc.4

**Comment by michaelstaib** [comment]
Thanks for reporting.

**Comment by michaelstaib** [comment]
Fixed with #5780
